### DISCUSSION TITLE
Rationalization of the use of harmonic electron-phonon coupling matrix

### DISCUSSION THREAD

**Discussion by ZhongyuWAN** [SSCHAcode/python-sscha#365] [Q&A] [answered]
Dear SSCHA developers,

I have a question regarding the fundamental principles of SSCHA and hope to receive your assistance. In Tutorial 7, "Calculation of the electron-phonon interaction and superconducting properties with the SSCHA," I noticed a step where the SSCHA dynamical matrix is preceded by the harmonic approximation dynamical matrix calculated using the QE package. This raised a question for me, considering the example of PdH, which is a strongly anharmonic crystal. Since the electron-phonon coupling matrix elements require both the dynamical matrix and the derivatives of the KS potential, while SSCHA can generate anharmonic dynamical matrices, can SSCHA also produce the derivatives of the KS potential? Quantum anharmonic effects can alter lattice and atomic displacements, so the derivatives of the anharmonic KS potential must differ from those under the harmonic approximation.

If the derivatives of the anharmonic KS potential can be computed by SSCHA, where are they stored? From the tutorial, I observed that a series of processes revolve around the harmonic electron-phonon coupling matrix elements. Is it reasonable to do so for a system with strong anharmonicity?

With regards,
Zhongyu

**Comment by DjordjeDangic** [comment, marked as answer]
Hello Zhongyu,

Please read the following paper in order to better understand electron-phonon coupling https://arxiv.org/abs/2303.02621 in these types of materials. This approach is not implemented in SSCHA.

Currently, we are treating electron-phonon as it is laid out in the tutorial you are mentioning. Is it reasonable? It is not less reasonable than purely harmonic calculation.

**Reply by ZhongyuWAN** [reply]
Hello, Djordje Dangic,

I truly appreciate you sharing this valuable literature with me, as it has greatly enhanced my understanding of EPC. However, it has also sparked more questions in my mind, and I hope to hear your answer to them.

As stated in this work focusing on the example of PdH, PdH is a conventional superconductor with significant nonlinear EPC due to its unusual isotope effect. My question is: How can one prove that no significant nonlinear EPC exists in a BCS superconductor? Is it a reasonable approach to verify that their isotope coefficients are close to the ideal value of 0.5? Especially considering the current SSCHA package cannot account for nonlinear EPC effects.

For instance, I conducted SSCHA calculations on a compound MHn, which exhibited an λ of 2.1. When I replaced H with its isotope D, the λ for MDn was 2.25. The estimated superconducting transition temperatures for both compounds approximately followed the isotope effect. Therefore, I wonder whether nonlinear EPC can be neglected in the MHn compound.

With regards,
Zhongyu

**Reply by DjordjeDangic** [reply]
Hello Zhongyu,

I don't think non-linear electron-phonon and isotope effect are directly related. Inverse isotope effect in PdH is related to the phonon frequency renormalization due to anharmonicity. The anharmonicity could be influenced by non-linear electron-phonon coupling, but not in an obvious way. 

I think it is safe to neglect non-linear electron-phonon. If you are really worried you could calculate non-linear electron-phonon by finite difference just as they did in the referenced paper.

**Reply by ZhongyuWAN** [reply]
Hi, Djordje Dangic,

Thank you very much for your prompt response. However, please forgive me for still having some confusion, as demonstrated in the SSCHA tutorial. The first step involves obtaining the electron-phonon coupling matrix elements under the harmonic approximation. However, after we perform NPT/NVT calculations using SSCHA, the optimized lattice from SSCHA differs from the initial harmonic lattice, either in terms of lattice constants or atomic occupations. Does this result in new electron-phonon coupling matrix elements? If so, is it reasonable to calculate anharmonic λ based on the initial harmonic electron-phonon coupling matrix elements? Such as https://sscha.eu/Tutorials/tutorial_07_simple_electron_phonon/

With regards,
Zhongyu

**Reply by DjordjeDangic** [reply]
Hello Zhongyu,

You should calculate electron-phonon for the new structure. Mixing phonons of one structure with electron-phonon matrix elements of another structure is not a consistent estimate of the electron-phonon interaction.
